phoenix5-spark/src/main/java/org/apache/phoenix/spark/datasource/v2/reader/PhoenixDataSourceReader.java

```diff
@@ -207,6 +208,27 @@ public Filter[] pushedFilters() {
 
     @Override
     public void pruneColumns(StructType schema) {
-        this.schema = schema;
+        if (schema.fields() != null && schema.fields().length != 0)
+            this.schema = schema;
+    }
+
+    //TODO Method PhoenixRuntime.generateColumnInfo skip only salt column, add skip tenant_id column.
+    private List<ColumnInfo> generateColumnInfo(Connection conn, String tableName) throws SQLException {
```

Comment:
You may want to consider exposing this functionality in PhoenixRuntime, or the Phoenix MR classes, if it is not yet exposed.

This is for later.

Reply:
I already created an issue for PhoenixRuntime.generateColumnInfo. I plan to submit a PR later.

Reply:
thanks.